### PR TITLE
feat: get user by id

### PR DESCRIPTION
**Pull request by khayss**
## What does the PR do?
This PR is about getting a user by Id

## How should this be manually tested?
Send a GET request to `/api/v1/users/:userId` with access token in the authorization header.

## Checklist of what you did
- [x] Implemented endpoint to get a user by the `user_id`.
- [x] Handle error for invalid user id in the params.
- [x] Integrate with auth guard to prevent request from unauthorized access.
- [x] Implement safety measure to avoid leak of user sensitive data such as password.
- [x] Requests to the endpoint must include a valid authentication token in the Authorization header.
- [x] Returned success `status_code ` with the user data.
- [x] Unit test the user service to ensure it works as expected.

## Test Screenshot

image

### CODE REVIEW

File: src/modules/user/user.controller.ts [hunk 1]
```diff
@@ -45,4 +57,32 @@ export class UserController {
...
+      return {
+        status_code: 200,
+        user: userData,
+      };
+    } catch (e) {
+      throw new HttpException(
+        {
+          status_code: 500,
+          error: 'Internal Server Error',
+        },
```
Comment (TheCodeGhinux) [inline src/modules/user/user.controller.ts:83]: why are you using only controller? This code should be in service file

**Comment by TheCodeGhinux** [comment]
Also, fix pr name, get user not "add get user"

File: src/modules/user/user.controller.ts (hunk 1 above)
Comment (khayss) [reply src/modules/user/user.controller.ts:83]: the service was already designed. I can't temper with it

**Comment by khayss** [comment]
The initial pull request was approved. same code. only had merge conflct later

File: src/modules/user/user.controller.ts [hunk 2]
```diff
@@ -1,9 +1,21 @@
...
+  Patch,
+  Param,
+  Body,
+  UsePipes,
+  ValidationPipe,
+  Request,
+  Req,
+  Get,
+  HttpException,
+} from '@nestjs/common';
```
Comment (davidolanrewaju) [inline src/modules/user/user.controller.ts:12]: Only import methods you are consuming. Removing all unnecessary methods from your import

**Pull request closed without merging**